src/System.Security.Cryptography.X509Certificates/src/Internal/Cryptography/Pal.Unix/DirectoryBasedStoreProvider.cs

```diff
@@ -0,0 +1,400 @@
+         
+            _storePath = Path.Combine(
+                Environment.GetEnvironmentVariable("HOME"),
+                ".microsoft",
+                "netfx",
+                "cryptography",
+                "x509stores",
+                directoryName);
+
+            if (0 != (openFlags & OpenFlags.OpenExistingOnly))
+            {
+                if (!Directory.Exists(_storePath))
+                {
+                    throw new CryptographicException(SR.Cryptography_X509_StoreNotFound);
+                }
+            }
+
+            // ReadOnly is 0x00, so it is implicit unless either ReadWrite or MaxAllowed
+            // was requested.
+            OpenFlags writeFlags = openFlags & (OpenFlags.ReadWrite | OpenFlags.MaxAllowed);
+
+            if (writeFlags == OpenFlags.ReadOnly)
+            {
+                _readOnly = true;
+            }
+        }
+        
+        public void Dispose()
+        {
+            if (_watcher != null)
+            {
+                _watcher.Dispose();
+                _watcher = null;
+            }
+        }
+
+        public void FindAndCopyTo(
+            X509FindType findType,
+            object findValue,
+            bool validOnly,
+            X509Certificate2Collection collection)
+        {
+        }
+
+        public byte[] Export(X509ContentType contentType, string password)
+        {
+            // Export is for X509Certificate2Collections in their IStorePal guise,
+            // if someone wanted to export whole stores they'd need to do
+            // store.Certificates.Export(...), which would end up in the
+            // CollectionBackedStoreProvider.
+            Debug.Fail("Export was unexpected on a DirectoryBasedStore");
+            throw new InvalidOperationException();
+        }
+
+        public void CopyTo(X509Certificate2Collection collection)
+        {
+            Debug.Assert(collection != null);
+
+            // Copy the reference locally, any directory change operations
+            // will cause the field to be reset to null.
+            List<X509Certificate2> certificates = _certificates;
+
+            if (certificates == null)
+            {
+                // ReadDirectory will both load _certificates and return the answer, so this call
+                // will have stable results across multiple adds/deletes happening in parallel.
+                certificates = ReadDirectory();
+                Debug.Assert(certificates != null);
+            }
+
+            foreach (X509Certificate2 cert in certificates)
+            {
+                collection.Add(cert);
+            }
+        }
+
+        private List<X509Certificate2> ReadDirectory()
+        {
+            if (!Directory.Exists(_storePath))
+            {
+                // Don't assign the field here, because we don't have a FileSystemWatcher
+                // yet to tell us that something has been added.
+                return new List<X509Certificate2>(0);
```

Comment:
> Though if you just mean make the line be akin to

Yeah, I was referring just to caching s_emptyList.  But if this isn't common, it's not worth the extra caching field.